core/src/main/kotlin/io/customer/sdk/communication/EventBus.kt

```diff
@@ -0,0 +1,56 @@
+package io.customer.sdk.communication
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.launch
+
+/**
+ * Event class to be used for communication between different modules.
+ */
+interface EventBus {
+    val flow: SharedFlow<Event>
+    fun publish(event: Event)
+    fun cancelAll()
+}
+
+/**
+ * Implementation of [EventBus] using [SharedFlow] for event handling.
+ * param scope: [CoroutineScope] to be used for event handling.
+ * param flow: [SharedFlow] to be used for event handling and buffer for replay.
+ */
+class EventBusImpl(
+    val scope: CoroutineScope = CoroutineScope(Dispatchers.Default + SupervisorJob()),
+    override val flow: SharedFlow<Event> = MutableSharedFlow(replay = 100)
+) : EventBus {
+
+    private val sharedFlow: MutableSharedFlow<Event>
+        get() = flow as MutableSharedFlow<Event>
```

Comment:
I wanted to ensure we can have control over `flow` so the constructor allows for any SharedFlow<Event> to be passed in, which can provide more flexibility if the implementation changes. But there is an typecasting overhead, I don't think there is going to be an issue with typecasting since type are defined already. 

So option 1, is out of contention. 

But I think option 2, somewhat also provides the ability to control flow, but is less flexible as it would `only` accept `MutableSharedFlow`. 

So I don't have any strong preference here, if you feel more strongly about option 2, I don't mind utilizing that.

Reply:
I would then prefer option 2 as casting is a bit unnecessary at the moment in my opinion.